core-services/chat/src/main/java/org/silverpeas/core/chat/servers/EJabberdServer.java

```diff
@@ -0,0 +1,173 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.chat.servers;
+
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.chat.ChatServerException;
+import org.silverpeas.core.chat.ChatUser;
+import org.silverpeas.core.util.JSONCodec;
+import org.silverpeas.core.util.SettingBundle;
+import org.silverpeas.core.util.logging.SilverLogger;
+
+import javax.ws.rs.core.Response;
+import java.util.function.Function;
+
+/**
+ * Adapter to a remote ejabberd service. It implements the mechanisms to access an ejabberd service
+ * in order to register users and to create networking relationships.
+ * @author mmoquillon
+ */
+@DefaultChatServer
+public class EJabberdServer implements ChatServer {
+
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  private final String url;
+
+  private final HttpRequester requester;
+
+  /**
+   * Constructs a new instance.
+   */
+  public EJabberdServer() {
+    SettingBundle settings = ChatServer.getChatSettings();
+    this.url = settings.getString("chat.xmpp.restUrl");
+    String key = settings.getString("chat.xmpp.restKey");
+    this.requester = new HttpRequester("Bearer " + key);
+  }
+
+  @Override
+  public void createUser(final User user) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Register user {0} ({1})", user.getDisplayedName(), user.getId());
+    ChatUser chatUser = ChatUser.fromUser(user);
+    request("register",
+        o -> o.put("user", chatUser.getChatLogin())
+        .put("host", chatUser.getChatDomain())
+        .put("password", chatUser.getChatPassword()));
+    request("set_vcard",
+        o -> o.put("user", chatUser.getChatLogin())
+        .put("host", chatUser.getChatDomain())
+        .put("name", "FN")
+        .put("content", chatUser.getDisplayedName()));
+  }
+
+  @Override
+  public void deleteUser(final User user) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Unregister user {0} ({1})", user.getDisplayedName(), user.getId());
+    ChatUser chatUser = ChatUser.fromUser(user);
+    request("unregister",
+        o -> o.put("username", chatUser.getChatLogin())
+            .put("hostname", chatUser.getChatDomain()));
+  }
+
+  @Override
+  public void createRelationShip(final User user1, final User user2) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Add relationships between {0} ({1}) and {2} ({3})", user1.getDisplayedName(),
+            user1.getId(), user2.getDisplayedName(), user2.getId());
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+    request("add_rosteritem",
+        o -> o.put("localuser", chatUser2.getChatLogin())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "localuser" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)